vm-validator/src/vm_validator.rs

```diff
@@ -225,10 +230,34 @@ impl<S: StateView> ModuleCodeBuilder for ValidationState<S> {
             Some(bytes) => bytes,
             None => return Ok(None),
         };
-        let compiled_module = self
+        let override_option_module = self
+            .environment
+            .runtime_environment()
+            .vm_config()
+            .override_option_module;
+        let enable_enum_option = self
+            .environment
             .runtime_environment()
-            .deserialize_into_compiled_module(state_value.bytes())?;
-        let extension = Arc::new(AptosModuleExtension::new(state_value));
```

Comment:
Maybe we can have this option override done by environment: deserialize into compiled module will give deserialize precipices bytes, so we handle this at one place?